docs/book/src/topics/crd-changes/v1alpha6-to-v1alpha7.md

```diff
@@ -111,3 +111,24 @@ The OpenStackMachine spec previously contained a `subnet` field which could used
 to set the `accessIPv4` field on Nova servers. This feature was not widely
 used, difficult to use, and could not be extended to support IPv6. It is
 removed without replacement.
+
+#### Removal of Port SecurityGroups
+
+The Port field of the OpenStackMachine spec previously contained both `securityGroups` and `securityGroupFilters`.
+As `securityGroups` can easily be replaced with `securityGroupFilters`, that can express the same and more, `securityGroups` has now been removed.
+CAPO can automatically convert `securityGroups` to `securityGroupFilters` when upgrading.
+
+Here is an example of how to use `securityGroupFilters` to replace `securityGroups`:
+
+```yaml
+# securityGroups are available in v1alpha6
+securityGroups:
+- 60ed83f1-8886-41c6-a1c7-fcfbdf3f04c2
+- 0ddd14d9-5c33-4734-b7d0-ac4fdf35c2d9
+- 4a131d3e-9939-4a6b-adea-788a2e89fcd8
+# securityGroupFilters are available in both v1alpha6 and v1alpha7
+securityGroupFilters:
+- UUID: 60ed83f1-8886-41c6-a1c7-fcfbdf3f04c2
+- UUID: 0ddd14d9-5c33-4734-b7d0-ac4fdf35c2d9
+- UUID: 4a131d3e-9939-4a6b-adea-788a2e89fcd8
```

Comment:
I think this should be:
```suggestion
- id: 60ed83f1-8886-41c6-a1c7-fcfbdf3f04c2
- id: 0ddd14d9-5c33-4734-b7d0-ac4fdf35c2d9
- id: 4a131d3e-9939-4a6b-adea-788a2e89fcd8
```

Incidentally, we should consider how to cleanup inconsistent use of id and uuid for OpenStack resources throughout the API. We should probably pick one and add it as an option everywhere it's not used, and mark the other one as deprecated.

Reply:
Good catch! I got so confused by this that I'm going to put what I found here for future reference.
The `PortOpts.SecurityGroupFilters` is `[]SecurityGroupParam`. This struct has the `UUID` field that translates to uuid in json/yaml, so that is what I should have put in the docs.
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/e35f1ca854b8bd7c3a06df8ff406aeb5f6c79d82/api/v1alpha7/types.go#L32-L39
To make things more confusing, the struct also contains a `Filter` of type `SecurityGroupFilter`. This is the one that has `ID` instead of `UUID`. I guess this is what you were talking about also? We could probably just use `SecurityGroupFilter` everywhere and get rid of the `SecurityGroupParam` struct.

Reply:
Updated `UUID -> uuid`